### ISSUE TITLE
Keyboard shortcuts for Git commands

### ISSUE THREAD

**Issue by uliska** [wbsoft/frescobaldi#940]
>>>  "Ctrl + z" undo operation support

>>    (Question: what do you mean with "undo" operation? If I'm not mistaken you haven't explained that yet in the email discussions)

>The idea of "undo" hit me when I was writing the Project Goals. While editing files in a editor, it's a natural way to press "Ctrl + Z" to undo the changes we've made. Frescobaldi has already support "undo" operations in the editor. I guess it's natively provided by PyQt.  

>I think if we can undo the operations of inline editors like "stage", "discard", "unstage" through pressing "Ctrl + Z", users would be more comfortable to use it. To undo "stage" operation, the editor reverts the staged lines into modified state. To undo "unstage" operation, the editor takes the discarded changes back.  And PyQt doesn't provide them natively. I don't know how to implement it till now (an undo stack?).  But it's worth a try.

I strongly recommend *not* to "hijack" such standardized keyboard shortcuts for different (albeit similar) meanings as this will thoroughly confuse users.

Of course functions like "stage current chunk" or "discard current chunk" that operate on the line where the cursor is in will have to get configurable keyboard shortcuts but `Ctrl+Z` should keep the usual meaning of "undo the most recent operation".

**Comment by wenxin-bupt** [comment]
I agree. The shortcuts thing is the detailed part of the function.  We can leave it until most functionalities have been implemented.

**Comment by uliska** [comment]
I think we should add two actions for now:

* show diff window for line/hunk
* revert current hunk (I suggest `Ctrl-Shift-Z` as default shortcut)

Both will be enabled when the cursor is in a line within a modified hunk.
This way they will be available from the `Version` menu and through keyboard shortcuts.

Basically this means we have to add VCS to the existing action/actioncollection stuff first, then add the actions and finally hook them to the appropriate places.

**Comment by uliska** [comment]
And I'm not sure, but maybe all the signals in the View should be piped through actions?

I hope (but am confident) that by now the codebase is VCS agnostic enough to easily support this regardless of the used VCS.

**Comment by uliska** [comment]
Question: *Where* should the actions be maintained, in the VCS module (e.g. vcs.actions.py) or in the view manager?

**Comment by uliska** [comment]
48ab6baae9498a5740e8c4eea65212165b8d2bfc redefines the organization of actions: From the commit message:

```
- Information about Frescobaldi branches
  is moved from vcs.menu to vcs.App (the former apprepo)
- The "Versioning" menu is created along the other menus in menu.py
- app.activeWindow().vcsManager maintains the action collection
- actions are defined in vcs.actions
```
